model/inline/src/main/scala/aqua/model/inline/raw/ApplyPropertiesRawInliner.scala

```diff
@@ -355,6 +355,29 @@ object ApplyPropertiesRawInliner extends RawInliner[ApplyPropertyRaw] with Loggi
               s"Unfolded stream ($vr) cannot be a literal"
             )
         }
+      case (
+            vr @ VarRaw(_, st @ StreamMapType(_)),
+            Some(iar @ IntoArrowRaw(arrowName, _, args), otherProperties)
+          ) =>
+        unfold(vr).flatMap {
+          case (VarModel(nameVM, _, _), inl) =>
+            for {
+              argsInlined <- args.traverse(unfold(_)).map(_.unzip)
+              (argsVM, argsInline) = argsInlined
+              mapInlined <- ApplyStreamMapRawInliner(arrowName, nameVM, st, argsVM)
+              (mapVM, mapInline) = mapInlined
```

Comment:
nit: Can you rename vars with `map` prefix to smth like `result`? It is confusing as this is not the map itself, but result of arrow application to the map.